Unit test validation_flush_tests fails on with recent libstdc++?

`validation_flush_tests/getcoinscachesizestate` fails on RISCV-64 on master:
```
Running 394 test cases...
test/validation_flush_tests.cpp(99): error: in "validation_flush_tests/getcoinscachesizestate": check chainstate.GetCoinsCacheSizeState(tx_pool, MAX_COINS_CACHE_BYTES, 0) == CoinsCacheSizeState::OK has failed [1 != 0]

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
```
Haven't investigated why, yet. Apart from the architecture it could also be some dependency version causing the problem (this is a fedora rawhide install with very new versions of everything including gcc and boost).

Edit: see below, this is likely due to a newer `libstdc++`

Same on debian:sid (ports)

```
make[4]: Entering directory '/home/test/bitcoin/build/bitcoin-riscv64-linux-gnu/src'
============================================================================
Testsuite summary for Bitcoin Core 0.19.99
============================================================================
# TOTAL: 0
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[4]: Leaving directory '/home/test/bitcoin/build/bitcoin-riscv64-linux-gnu/src'
Running tests: addrman_tests from test/addrman_tests.cpp
Running tests: amount_tests from test/amount_tests.cpp
Running tests: allocator_tests from test/allocator_tests.cpp
Running tests: base32_tests from test/base32_tests.cpp
Running tests: base58_tests from test/base58_tests.cpp
Running tests: base64_tests from test/base64_tests.cpp
Running tests: bech32_tests from test/bech32_tests.cpp
Running tests: bip32_tests from test/bip32_tests.cpp
Running tests: blockchain_tests from test/blockchain_tests.cpp
Running tests: blockencodings_tests from test/blockencodings_tests.cpp
Running tests: blockfilter_tests from test/blockfilter_tests.cpp
Running tests: blockfilter_index_tests from test/blockfilter_index_tests.cpp
Running tests: bloom_tests from test/bloom_tests.cpp
Running tests: bswap_tests from test/bswap_tests.cpp
Running tests: checkqueue_tests from test/checkqueue_tests.cpp
Running tests: coins_tests from test/coins_tests.cpp
Running tests: compilerbug_tests from test/compilerbug_tests.cpp
Running tests: compress_tests from test/compress_tests.cpp
Running tests: crypto_tests from test/crypto_tests.cpp
Running tests: cuckoocache_tests from test/cuckoocache_tests.cpp
Running tests: denialofservice_tests from test/denialofservice_tests.cpp
Running tests: descriptor_tests from test/descriptor_tests.cpp
Running tests: flatfile_tests from test/flatfile_tests.cpp
Running tests: fs_tests from test/fs_tests.cpp
Running tests: getarg_tests from test/getarg_tests.cpp
Running tests: hash_tests from test/hash_tests.cpp
Running tests: key_io_tests from test/key_io_tests.cpp
Running tests: key_tests from test/key_tests.cpp
Running tests: limitedmap_tests from test/limitedmap_tests.cpp
Running tests: logging_tests from test/logging_tests.cpp
Running tests: dbwrapper_tests from test/dbwrapper_tests.cpp
Running tests: validation_tests from test/validation_tests.cpp
Running tests: mempool_tests from test/mempool_tests.cpp
Running tests: merkle_tests from test/merkle_tests.cpp
Running tests: merkleblock_tests from test/merkleblock_tests.cpp
Running tests: miner_tests from test/miner_tests.cpp
Running tests: multisig_tests from test/multisig_tests.cpp
Running tests: net_tests from test/net_tests.cpp
Running tests: netbase_tests from test/netbase_tests.cpp
Running tests: pmt_tests from test/pmt_tests.cpp
Running tests: policyestimator_tests from test/policyestimator_tests.cpp
Running tests: pow_tests from test/pow_tests.cpp
Running tests: prevector_tests from test/prevector_tests.cpp
Running tests: raii_event_tests from test/raii_event_tests.cpp
Running tests: random_tests from test/random_tests.cpp
Running tests: reverselock_tests from test/reverselock_tests.cpp
Running tests: rpc_tests from test/rpc_tests.cpp
Running tests: sanity_tests from test/sanity_tests.cpp
Running tests: scheduler_tests from test/scheduler_tests.cpp
Running tests: script_p2sh_tests from test/script_p2sh_tests.cpp
Running tests: script_tests from test/script_tests.cpp
Running tests: script_standard_tests from test/script_standard_tests.cpp
Running tests: scriptnum_tests from test/scriptnum_tests.cpp
Running tests: serialize_tests from test/serialize_tests.cpp
Running tests: settings_tests from test/settings_tests.cpp
Running tests: sighash_tests from test/sighash_tests.cpp
Running tests: sigopcount_tests from test/sigopcount_tests.cpp
Running tests: skiplist_tests from test/skiplist_tests.cpp
Running tests: streams_tests from test/streams_tests.cpp
Running tests: sync_tests from test/sync_tests.cpp
Running tests: util_threadnames_tests from test/util_threadnames_tests.cpp
Running tests: timedata_tests from test/timedata_tests.cpp
Running tests: torcontrol_tests from test/torcontrol_tests.cpp
Running tests: transaction_tests from test/transaction_tests.cpp
Running tests: txindex_tests from test/txindex_tests.cpp
Running tests: txvalidation_tests from test/txvalidation_tests.cpp
Running tests: txvalidationcache_tests from test/txvalidationcache_tests.cpp
Running tests: uint256_tests from test/uint256_tests.cpp
Running tests: util_tests from test/util_tests.cpp
Running tests: validation_block_tests from test/validation_block_tests.cpp
Running tests: validation_flush_tests from test/validation_flush_tests.cpp
Running 1 test case...
Test cases order is shuffled using seed: 902477190
Entering test module "Bitcoin Core Test Suite"
test/validation_flush_tests.cpp(12): Entering test suite "validation_flush_tests"
test/validation_flush_tests.cpp(19): Entering test case "getcoinscachesizestate"
2020-02-11T00:10:09Z Seed: Setting random seed for current tests to RANDOM_CTX_SEED=96807d9fd7f28bc0da6041aeaf760a509ee36b39cfa80ed6d37f4abb89713310
2020-02-11T00:10:09Z Bitcoin Core version v0.19.99.0-55ece5fe2-dirty (release build)
2020-02-11T00:10:09Z Using 16 MiB out of 32/2 requested for signature cache, able to store 524288 elements
2020-02-11T00:10:09Z Using 16 MiB out of 32/2 requested for script execution cache, able to store 524288 elements
2020-02-11T00:10:09Z Opened LevelDB successfully
2020-02-11T00:10:09Z Wrote new obfuscate key for /tmp/test_common_Bitcoin Core/3944085796/chainstate: 2f91067c72122910
2020-02-11T00:10:09Z Using obfuscation key for /tmp/test_common_Bitcoin Core/3944085796/chainstate: 2f91067c72122910
CCoinsViewCache memory usage: 32
CCoinsViewCache memory usage: 336
CCoinsViewCache memory usage: 544
CCoinsViewCache memory usage: 752
CCoinsViewCache memory usage: 960
test/validation_flush_tests.cpp(99): error: in "validation_flush_tests/getcoinscachesizestate": check chainstate.GetCoinsCacheSizeState(tx_pool, MAX_COINS_CACHE_BYTES, 0) == CoinsCacheSizeState::OK has failed [1 != 0]
CCoinsViewCache memory usage: 1168
2020-02-11T00:10:09Z Cache size (1168) exceeds total space (1024)
2020-02-11T00:10:09Z Cache size (1168) exceeds total space (1024)
CCoinsViewCache memory usage: 1376
CCoinsViewCache memory usage: 1584
CCoinsViewCache memory usage: 1792
CCoinsViewCache memory usage: 2000
CoinsTip usage percentage: 0.976562
2020-02-11T00:10:09Z Cache size (218768) exceeds total space (1024)
CCoinsViewCache memory usage: 8896
2020-02-11T00:10:09Z Cache size (8896) exceeds total space (1024)
test/validation_flush_tests.cpp(19): Leaving test case "getcoinscachesizestate"; testing time: 204280us
test/validation_flush_tests.cpp(12): Leaving test suite "validation_flush_tests"; testing time: 205321us
Leaving test module "Bitcoin Core Test Suite"; testing time: 206834us

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
make[3]: *** [Makefile:17898: test/validation_flush_tests.cpp.test] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/home/test/bitcoin/build/bitcoin-riscv64-linux-gnu/src'
make[2]: *** [Makefile:16441: check-am] Error 2
make[2]: Leaving directory '/home/test/bitcoin/build/bitcoin-riscv64-linux-gnu/src'
make[1]: *** [Makefile:16126: check-recursive] Error 1
make[1]: Leaving directory '/home/test/bitcoin/build/bitcoin-riscv64-linux-gnu/src'
make: *** [Makefile:790: check-recursive] Error 1

--- a/src/test/bswap_tests.cpp
+++ b/src/test/bswap_tests.cpp
@@ -11,16 +11,16 @@ BOOST_FIXTURE_TEST_SUITE(bswap_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(bswap_tests)
 {
-	// Sibling in bitcoin/src/qt/test/compattests.cpp
-	uint16_t u1 = 0x1234;
-	uint32_t u2 = 0x56789abc;
-	uint64_t u3 = 0xdef0123456789abc;
-	uint16_t e1 = 0x3412;
-	uint32_t e2 = 0xbc9a7856;
-	uint64_t e3 = 0xbc9a78563412f0de;
-	BOOST_CHECK(bswap_16(u1) == e1);
-	BOOST_CHECK(bswap_32(u2) == e2);
-	BOOST_CHECK(bswap_64(u3) == e3);
+    // Sibling in bitcoin/src/qt/test/compattests.cpp
+    uint16_t u1 = 0x1234;
+    uint32_t u2 = 0x56789abc;
+    uint64_t u3 = 0xdef0123456789abc;
+    uint16_t e1 = 0x3412;
+    uint32_t e2 = 0xbc9a7856;
+    uint64_t e3 = 0xbc9a78563412f0de;
+    BOOST_CHECK(bswap_16(u1) == e1);
+    BOOST_CHECK(bswap_32(u2) == e2);
+    BOOST_CHECK(bswap_64(u3) == e3);
 }
 
 BOOST_AUTO_TEST_SUITE_END()

--- a/src/test/transaction_tests.cpp
+++ b/src/test/transaction_tests.cpp
@@ -122,10 +122,9 @@ BOOST_AUTO_TEST_CASE(tx_valid)
             std::map<COutPoint, int64_t> mapprevOutValues;
             UniValue inputs = test[0].get_array();
             bool fValid = true;
-	    for (unsigned int inpIdx = 0; inpIdx < inputs.size(); inpIdx++) {
-	        const UniValue& input = inputs[inpIdx];
-                if (!input.isArray())
-                {
+            for (unsigned int inpIdx = 0; inpIdx < inputs.size(); inpIdx++) {
+                const UniValue& input = inputs[inpIdx];
+                if (!input.isArray()) {
                     fValid = false;
                     break;
                 }
@@ -209,10 +208,9 @@ BOOST_AUTO_TEST_CASE(tx_invalid)
             std::map<COutPoint, int64_t> mapprevOutValues;
             UniValue inputs = test[0].get_array();
             bool fValid = true;
-	    for (unsigned int inpIdx = 0; inpIdx < inputs.size(); inpIdx++) {
-	        const UniValue& input = inputs[inpIdx];
-                if (!input.isArray())
-                {
+            for (unsigned int inpIdx = 0; inpIdx < inputs.size(); inpIdx++) {
+                const UniValue& input = inputs[inpIdx];
+                if (!input.isArray()) {
                     fValid = false;
                     break;
                 }

--- a/src/test/validation_flush_tests.cpp
+++ b/src/test/validation_flush_tests.cpp
@@ -2,10 +2,10 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 //
-#include <txmempool.h>
-#include <validation.h>
 #include <sync.h>
 #include <test/util/setup_common.h>
+#include <txmempool.h>
+#include <validation.h>
 
 #include <boost/test/unit_test.hpp>
 
@@ -85,12 +85,7 @@ BOOST_AUTO_TEST_CASE(getcoinscachesizestate)
     // This is contingent not only on the dynamic memory usage of the Coins
     // that we're adding (COIN_SIZE bytes per), but also on how much memory the
     // cacheCoins (unordered_map) preallocates.
-    //
-    // I came up with the count by examining the printed memory usage of the
-    // CCoinsCacheView, so it's sort of arbitrary - but it shouldn't change
-    // unless we somehow change the way the cacheCoins map allocates memory.
-    //
-    constexpr int COINS_UNTIL_CRITICAL = is_64_bit ? 4 : 5;
+    constexpr int COINS_UNTIL_CRITICAL{3};
 
     for (int i{0}; i < COINS_UNTIL_CRITICAL; ++i) {
         COutPoint res = add_coin(view);
@@ -101,17 +96,14 @@ BOOST_AUTO_TEST_CASE(getcoinscachesizestate)
             CoinsCacheSizeState::OK);
     }
 
-    // Adding an additional coin will push us over the edge to CRITICAL.
-    add_coin(view);
-    print_view_mem_usage(view);
-
-    auto size_state = chainstate.GetCoinsCacheSizeState(
-        tx_pool, MAX_COINS_CACHE_BYTES, /*max_mempool_size_bytes*/ 0);
-
-    if (!is_64_bit && size_state == CoinsCacheSizeState::LARGE) {
-        // On 32 bit hosts, we may hit LARGE before CRITICAL.
+    // Adding some additional coins will push us over the edge to CRITICAL.
+    for (int i{0}; i < 4; ++i) {
         add_coin(view);
         print_view_mem_usage(view);
+        if (chainstate.GetCoinsCacheSizeState(tx_pool, MAX_COINS_CACHE_BYTES, /*max_mempool_size_bytes*/ 0) ==
+            CoinsCacheSizeState::CRITICAL) {
+            break;
+        }
     }
 
     BOOST_CHECK_EQUAL(

